app/modules/parsing/graph_construction/code_graph_service.py

```
@@ -18,6 +18,43 @@
+    )
+
+    # create_graph() only dispatches to the rust/python graph builders, so
+    # the token-counter/io collaborators (removed on this base) aren't needed.
+    repo_map = RepoMap(root=repo_dir, verbose=True)
+    parse_start = time.time()
+    graph = repo_map.create_graph(repo_dir)
+    parse_time = time.time() - parse_start
+    logger.info(
+        "[STRUCTURE PARSING] Parsed repository: "
+        f"{graph.number_of_nodes()} nodes, "
+        f"{graph.number_of_edges()} relationships in {parse_time:.2f}s",
```

Ruff (0.16.0)

[warning] Logging statement uses f-string

(G004)